workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -19,7 +20,40 @@ namespace <#= qualifiedNamespace #>
         {
             public uint ComponentId => <#= componentDetails.ComponentId #>;
 
-            public BlittableBool DirtyBit { get; set; }
+            [Flags]
+            public enum DirtyBitsEnum
+            {
+                None = 0,
+<# int currentFieldBinaryIdx = 1; #>
+<# foreach(var fieldDetails in fieldDetailsList) { #>
+                <#= fieldDetails.PascalCaseName #> = <#= currentFieldBinaryIdx #>,
+<# currentFieldBinaryIdx = currentFieldBinaryIdx << 1; #>
+<# } #>
+            }
+
+            private DirtyBitsEnum dirtyBits;
+
+            public DirtyBitsEnum DirtyBits => dirtyBits;
+
+            private void AddDirtyBit(DirtyBitsEnum currentField)
```

Comment:
These functions will be called a lot, but only from within generated code. Will be worth inlining them! Its generated code anyways, no harm in having it be less readable but faster.